### PR TITLE
Organize targets properly in folders

### PR DESCRIPTION
**Pull request by tribal-tec**
FYI @rdumusc

### CODE REVIEW

File: DoxygenRule.cmake [hunk 1]
```diff
@@ -80,35 +80,47 @@ if(NOT DOXYGEN_CONFIG_FILE)
...
-add_dependencies(doxygen_html ${PROJECT_NAME}_doxygen_html)
+add_dependencies(doxygen_html doxygen_html_${PROJECT_NAME})
 
-add_custom_target(${PROJECT_NAME}_doxygen DEPENDS ${PROJECT_NAME}_doxygen_html)
+add_custom_target(doxygen_${PROJECT_NAME} DEPENDS doxygen_html_${PROJECT_NAME})
+set_target_properties(doxygen_${PROJECT_NAME} PROPERTIES
+  EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER "doxygen")
 if(COVERAGE) # CoverageReport generated by "tests" in this case
-  add_dependencies(${PROJECT_NAME}_doxygen tests)
+  add_dependencies(doxygen tests_${PROJECT_NAME})
```
Comment (rdumusc) [inline DoxygenRule.cmake]: Should be doxygen_${PROJECT_NAME} tests_${PROJECT_NAME}

**Comment by rdumusc** [comment]
+1, this improves consistency

File: DoxygenRule.cmake (hunk 1 above)
Comment (tribal-tec) [inline DoxygenRule.cmake]: Oh, good catch!

**Comment by tribal-tec** [comment]
Yes, it was really messy in Visual Studio. Now waaaay cleaner :)

**Comment by eile** [comment]
+1

**Pull request merged**